src/main/java/com/synopsys/integration/alert/channel/azure/boards/web/AzureBoardsCustomEndpoint.java

```diff
@@ -22,37 +22,118 @@
+    protected OAuthEndpointResponse createOAuthResponse(FieldModel fieldModel, HttpServletContentWrapper servletContentWrapper) {
+        try {
+            FieldAccessor fieldAccessor = createFieldAccessor(fieldModel);
+            Optional<String> clientId = fieldAccessor.getString(AzureBoardsDescriptor.KEY_CLIENT_ID);
+            if (!clientId.isPresent()) {
+                return new OAuthEndpointResponse(HttpStatus.BAD_REQUEST.value(), false, "", "client id not found.");
+            }
+            Optional<String> alertServerUrl = alertProperties.getServerUrl();
+
+            if (!alertServerUrl.isPresent()) {
+                return new OAuthEndpointResponse(HttpStatus.BAD_REQUEST.value(), false, "", "Could not determine the alert server url for the callback.");
+            }
+            String authUrl = createAuthURL(clientId.get(), alertServerUrl.get());
+            logger.info("Authenticating Azure OAuth URL: " + authUrl);
+            //TODO add code to check if Alert has already been authorized to set the authenticated flag.
+
+            return new OAuthEndpointResponse(HttpStatus.OK.value(), false, authUrl, "");
+
+        } catch (Exception ex) {
+            logger.error("Error activating Azure Boards", ex);
+            return new OAuthEndpointResponse(HttpStatus.INTERNAL_SERVER_ERROR.value(), false, "", "Error activating azure oauth.");
+        }
+    }
+
+    private FieldAccessor createFieldAccessor(FieldModel fieldModel) {
+        Map<String, ConfigurationFieldModel> fields = new HashMap<>();
+        try {
+            fields.putAll(modelConverter.convertToConfigurationFieldModelMap(fieldModel));
+            // check if a configuration exists because the client id is a sensitive field and won't have a value in the field model if updating.
+            if (StringUtils.isNotBlank(fieldModel.getId())) {
+                Optional<ConfigurationModel> configurationFieldModel = configurationAccessor.getConfigurationById(Long.valueOf(fieldModel.getId()));
+                configurationFieldModel.ifPresent(model -> fields.putAll(model.getCopyOfKeyToFieldMap()));
+            }
+        } catch (AlertDatabaseConstraintException ex) {
+            logger.error("Error creating field acessor for Azure authentication", ex);
+        }
+        return new FieldAccessor(fields);
     }
 
+    private String createAuthURL(String clientId, String alertServerUrl) {
+        StringBuilder authUrlBuilder = new StringBuilder(300);
+        authUrlBuilder.append(AzureHttpServiceFactory.DEFAULT_AUTHORIZATION_URL);
+        authUrlBuilder.append(createQueryString(clientId, alertServerUrl));
+        return authUrlBuilder.toString();
+    }
+
+    private String createQueryString(String clientId, String alertServerUrl) {
+        String authorizationUrl = String.format("%s%s", alertServerUrl, AzureOauthCallbackController.AZURE_OAUTH_CALLBACK_PATH);
+        StringBuilder queryBuilder = new StringBuilder(250);
```

Comment:
Same thought here. If this is the length requirement, we should be more explicit about it.

Reply:
The 250 isn't a max length.  It's a best guess for initial capacity.  My test URL was 237 characters long. This is so the StringBuilder allocates a string with enough capacity initially instead of resizing more than once with the append calls.  it will resize dynamically like an ArrayList.  So this just makes it perform a little better.  The string underneath is backed by a character array.  So the character array will be created with a good enough size to avoid resizing during the append calls.